delivery_internal/models/stock_pickup_request.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class StockPickupRequest(models.Model):
+    _name = 'stock.pickup.request'
+    _description = 'Stock Pickup Request'
+
+    name = fields.Char(
+        string='Pickup Name',
+        default=lambda s: s._default_name(),
+    )
+    company_id = fields.Many2one(
+        string='Responsible Company',
+        comodel_name='res.company',
+        required=True,
+    )
+    cash_on_delivery = fields.Boolean()
+    state = fields.Selection(
+        selection=[
+            ('new', 'New'),
+            ('confirmed', 'Confirmed'),
+            ('done', 'done')
+        ],
+        compute='_compute_state',
+        store=True,
+    )
+    picking_id = fields.Many2one(
+        string='Source Picking',
+        comodel_name='stock.picking',
+        required=True,
+    )
+    in_picking_id = fields.Many2one(
+        string='Inbound Buyer Picking',
+        comodel_name='stock.picking',
+    )
+    out_picking_id = fields.Many2one(
+        string='Carrier Pickup Picking',
+        comodel_name='stock.picking',
+    )
+    cash_in_picking_id = fields.Many2one(
+        string='Cash Picking From Buyer Picking',
+        comodel_name='stock.picking',
+    )
+    cash_out_picking_id = fields.Many2one(
+        string='Cash Delivery From Carrier Picking',
+        comodel_name='stock.picking',
+    )
+
+    def _get_pick_type(self, picking, op_type):
+        """ It returns the picking type for the given picking's carrier
+        Args:
+            picking (stock.picking): The parent picking to use
+            op_type (str): Whether we need to stock or cash pick type
+        Returns:
+            stock.picking.type: The picking type
+        """
+        return getattr(
+            picking.carrier_id,
+            'internal_delivery_%s_picking_type_id' % op_type
+        )
+
+    def _create_move_lines(self, picking):
+        """ It will create move lines based on the given picking
```

Comment:
@tedsalmon IMO this method should be renamed since it doesn't create at all the move lines but prepare the data to create the move lines.